### ISSUE TITLE
Can't use eslint_d in multiple projects

### ISSUE THREAD

**Issue by DamienCassou** [mantoni/eslint_d.js#296]
I have eslint_d working in 1 project. When trying to use it in another one, I get:

```
$ eslint_d src/app/Content.jsx 
node:net:1297
    validatePort(port);
    ^

RangeError [ERR_SOCKET_BAD_PORT]: Port should be >= 0 and < 65536. Received type number (NaN).
    at lookupAndConnect (node:net:1297:5)
    at Socket.connect (node:net:1254:5)
    at Object.connect (node:net:237:17)
    at forwardToDaemon (file:///nix/store/727cpwihd52lvn098wpqhar4r322pgwq-eslint_d-14.0.0/lib/node_modules/eslint_d/lib/forwarder.js:18:22)
    at file:///nix/store/727cpwihd52lvn098wpqhar4r322pgwq-eslint_d-14.0.0/lib/node_modules/eslint_d/bin/eslint_d.js:64:13 {
  code: 'ERR_SOCKET_BAD_PORT'
}

Node.js v20.14.0
```

I'm using eslint_d v14.

**Comment by mantoni** [comment]
Can you send the output of `cat node_modules/eslint/.eslint_d` in the project where it doesn't work?

**Comment by DamienCassou** [comment]
The 2 distinct NPM packages share their `node_modules/` folder because they are part of the same monorepo. So the structure is:

- project/
  -  package.json     `# with a "workspaces: ["apps/*"]" entry`
  - node_modules/
    - eslint/
  - apps/app1/
    - package.json
    - eslint.config.js
    - node_modules/   `# with no eslint inside`
  - apps/app2/
    - package.json
    - eslint.config.js
    - node_modules/   `# with no eslint inside`

**Comment by mantoni** [comment]
Ok, so one issue I found is that the `hash` is created from files in the `cwd` rather than relative to the resolved base directory. I made a fix in https://github.com/mantoni/eslint_d.js/pull/297.

I'd still be interested in whether there is `undefined` or something weird as the port number in `project/node_modules/eslint/.eslint_d`. 🤔

**Comment by DamienCassou** [comment]
```
$ cat ../../node_modules/eslint/.eslint_d 
393e3985501e0f53 37773 115143 3ssdbUvZot16udWgBw/asw==
```

**Comment by DamienCassou** [comment]
PR #297 seems to fix the issue for me. Feel free to close this issue.

Do you expect tools to search for the `node_modules/eslint/.eslint_d` file and read that to get the port?

**Comment by mantoni** [comment]
> Do you expect tools to search for the node_modules/eslint/.eslint_d file and read that to get the port?

No.

**Comment by mantoni** [comment]
📦 Released `v14.0.1` with the fix.
